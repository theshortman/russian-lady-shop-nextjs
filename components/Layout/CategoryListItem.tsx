import { FC } from "react";
import Link from "next/link";

interface CategoryListItemProps {
  name: string;
  slug: string;
}

const CategoryListItem: FC<CategoryListItemProps> = ({ name, slug }) => {
  return (
    <Link href={`/catalog/${slug}`}>
      <a className="list-group-item list-group-item-action" aria-current="page">
        {name}
      </a>
    </Link>
  );
};

export default CategoryListItem;
