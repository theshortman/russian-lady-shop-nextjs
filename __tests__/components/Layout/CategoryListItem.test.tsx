import { render } from "@testing-library/react";
import CategoryListItem from "@/components/Layout/CategoryListItem";

test("renders CategoryListItem unchanged", () => {
  const category = { name: "Test category", slug: "test-slug" };

  const { container } = render(<CategoryListItem {...category} />);
  expect(container).toMatchSnapshot();
});
