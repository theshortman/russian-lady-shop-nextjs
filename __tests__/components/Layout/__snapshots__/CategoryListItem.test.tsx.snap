// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CategoryListItem unchanged 1`] = `
<div>
  <a
    aria-current="page"
    class="list-group-item list-group-item-action"
    href="/catalog/test-slug"
  >
    Test category
  </a>
</div>
`;
